### ISSUE TITLE
python: src/../include/data.cuh:34: void CPUData::assertDimensions(): Assertion `_data->at(i-1)->isTrans() == _data->at(i)->isTrans()' failed.

### ISSUE THREAD

**Issue by cmxnono** [akrizhevsky/cuda-convnet2#18]
python convnet.py --data-path ~/data/8_imagenet/ilsvrc2012/batchesx --train-range 0-417 --test-range 1000-1016 --save-path 
~/data/8_imagenet/ilsvrc2012/models/  --epochs 90 --layer-def layers/layers-imagenet-1gpu.cfg --layer-params layers/layer-params-imagenet-1gpu.cfg --da
ta-provider image --inner-size 227 --gpu 0 --mini 128 --test-freq 201 --color-noise 0.1
Initialized data layer 'data', producing 154587 outputs
Initialized data layer 'labvec', producing 1 outputs
Initialized convolutional layer 'conv1' on GPUs 0, producing 55x55 64-channel output
Initialized cross-map response-normalization layer 'rnorm1' on GPUs 0, producing 55x55 64-channel output
Initialized max-pooling layer 'pool1' on GPUs 0, producing 27x27 64-channel output
Initialized convolutional layer 'conv2' on GPUs 0, producing 27x27 192-channel output
Initialized cross-map response-normalization layer 'rnorm2' on GPUs 0, producing 27x27 192-channel output
Initialized max-pooling layer 'pool2' on GPUs 0, producing 13x13 192-channel output
Initialized convolutional layer 'conv3' on GPUs 0, producing 13x13 384-channel output
Initialized convolutional layer 'conv4' on GPUs 0, producing 13x13 256-channel output
Initialized convolutional layer 'conv5' on GPUs 0, producing 13x13 256-channel output
Initialized max-pooling layer 'pool3' on GPUs 0, producing 6x6 256-channel output
Initialized fully-connected layer 'fc4096a' on GPUs 0, producing 4096 outputs
Initialized dropout2 layer 'dropout1' on GPUs 0, producing 4096 outputs
Initialized fully-connected layer 'fc4096b' on GPUs 0, producing 4096 outputs
Initialized dropout2 layer 'dropout2' on GPUs 0, producing 4096 outputs
Initialized fully-connected layer 'fc1000' on GPUs 0, producing 1000 outputs
Initialized softmax layer 'probs' on GPUs 0, producing 1000 outputs
Initialized logistic regression cost 'logprob' on GPUs 0
Initialized neuron layer 'fc4096b_neuron' on GPUs 0, producing 4096 outputs
Initialized neuron layer 'conv3_neuron' on GPUs 0, producing 64896 outputs
Initialized neuron layer 'conv2_neuron' on GPUs 0, producing 139968 outputs
Initialized neuron layer 'conv4_neuron' on GPUs 0, producing 43264 outputs
Initialized neuron layer 'pool3_neuron' on GPUs 0, producing 9216 outputs
Initialized neuron layer 'pool1_neuron' on GPUs 0, producing 46656 outputs
Initialized neuron layer 'fc4096a_neuron' on GPUs 0, producing 4096 outputs
Layer conv3_neuron using acts from layer conv3
Layer fc4096a_neuron using acts from layer fc4096a
Layer fc4096b_neuron using acts from layer fc4096b
Layer conv2_neuron using acts from layer conv2
Layer conv4_neuron using acts from layer conv4
=========================
Importing cudaconvnet._ConvNet C++ module
Fwd terminal: logprob
found bwd terminal conv1[0] in passIdx=0
=========================
Training ConvNet
Add PCA noise to color channels with given scale                        : 0.1 
Check gradients and quit?                                               : 0  [DEFAULT]
Conserve GPU memory (slower)?                                           : 0  [DEFAULT]
Convert given conv layers to unshared local                             :    
Cropped DP: crop size (0 = don't crop)                                  : 227 
Cropped DP: test on multiple patches?                                   : 0  [DEFAULT]
Data batch range: testing                                               : 1000-1016 
Data batch range: training                                              : 0-417 
Data path                                                               : ./data/8_imagenet/ilsvrc2012/batchesx 
Data provider                                                           : image 
Force save before quitting                                              : 0  [DEFAULT]
GPU override                                                            : 0  
Layer definition file                                                   : layers/layers-imagenet-1gpu.cfg 
Layer file path prefix                                                  :    [DEFAULT]
Layer parameter file                                                    : layers/layer-params-imagenet-1gpu.cfg 
Load file                                                               :    [DEFAULT]
Logreg cost layer name (for --test-out)                                 :    [DEFAULT]
Minibatch size                                                          : 128 
Number of epochs                                                        : 90 
Output test case predictions to given path                              :    [DEFAULT]
Save file override                                                      :    
Save path                                                               : ./data/8_imagenet/ilsvrc2012/models/ 
Subtract this scalar from image (-1 = don't)                            : -1 [DEFAULT]
Test and quit?                                                          : 0  [DEFAULT]
Test on one batch at a time?                                            : 1  [DEFAULT]
Testing frequency                                                       : 201 
Unshare weight matrices in given layers                                 :    
Write test data features from given layer                               :    [DEFAULT]
Write test data features to this path (to be used with --write-features):    [DEFAULT]
=========================
Running on CUDA device(s) 0
Current time: Fri Jun 12 00:03:57 2015
Saving checkpoints to ./data/8_imagenet/ilsvrc2012/models/ConvNet__2015-06-12_00.03.53
=========================
1.0 (0.00%)... FUCK1
FUCK11
python: src/../include/data.cuh:34: void CPUData::assertDimensions(): Assertion `_data->at(i-1)->isTrans() == _data->at(i)->isTrans()' failed.
Error signal 6:
/home/chenxiu/src/36_convnet/cudaconvnet/_ConvNet.so(_Z13signalHandleri+0x1b)[0x7fc1c442d5fb]
/lib64/libc.so.6[0x37c6a326a0]
/lib64/libc.so.6(gsignal+0x35)[0x37c6a32625]
/lib64/libc.so.6(abort+0x175)[0x37c6a33e05]
/lib64/libc.so.6[0x37c6a2b74e]
/lib64/libc.so.6(__assert_perror_fail+0x0)[0x37c6a2b810]
./src/36_convnet/cudaconvnet/_ConvNet.so(_ZN7CPUData16assertDimensionsEv+0x144)[0x7fc1c442eb34]
./src/36_convnet/cudaconvnet/_ConvNet.so(_Z10startBatchP7_objectS0_+0x78)[0x7fc1c442dee8]
/usr/lib64/libpython2.6.so.1.0(PyEval_EvalFrameEx+0x5244)[0x37d3ad59e4]
/usr/lib64/libpython2.6.so.1.0(PyEval_EvalFrameEx+0x63ef)[0x37d3ad6b8f]
/usr/lib64/libpython2.6.so.1.0(PyEval_EvalCodeEx+0x927)[0x37d3ad7657]
/usr/lib64/libpython2.6.so.1.0(PyEval_EvalFrameEx+0x5304)[0x37d3ad5aa4]
/usr/lib64/libpython2.6.so.1.0(PyEval_EvalFrameEx+0x63ef)[0x37d3ad6b8f]
/usr/lib64/libpython2.6.so.1.0(PyEval_EvalFrameEx+0x63ef)[0x37d3ad6b8f]
/usr/lib64/libpython2.6.so.1.0(PyEval_EvalCodeEx+0x927)[0x37d3ad7657]
/usr/lib64/libpython2.6.so.1.0(PyEval_EvalCode+0x32)[0x37d3ad7732]
/usr/lib64/libpython2.6.so.1.0[0x37d3af1bac]
/usr/lib64/libpython2.6.so.1.0(PyRun_FileExFlags+0x90)[0x37d3af1c80]
/usr/lib64/libpython2.6.so.1.0(PyRun_SimpleFileExFlags+0xdc)[0x37d3af316c]
/usr/lib64/libpython2.6.so.1.0(Py_Main+0xb62)[0x37d3aff8a2]
/lib64/libc.so.6(__libc_start_main+0xfd)[0x37c6a1ed5d]
python[0x400649]

**Comment by cmxnono** [comment]
Can anyone help me?

**Comment by cmxnono** [comment]
I found(http://docs.scipy.org/doc/numpy-1.7.0/reference/generated/numpy.ndarray.T.html), when the dim of matrix is < 2,the isTrans() tag won't be set to true;
The method is to disable the assert，now it work normally！！
